### PR TITLE
fix using defer in loop in cors test

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What this PR does / why we need it**:
fix bug in cors test

**Special notes for your reviewer**:
a small bug fix, wrap defer in a closure
**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/server/filters/cors_test.go [hunk 1]
```diff
@@ -42,21 +42,23 @@ func TestCORSAllowedOrigins(t *testing.T) {
...
-			t.Errorf("unexpected error: %v", err)
-		}
-		request.Header.Set("Origin", item.origin)
-
-		response, err := client.Do(request)
-		if err != nil {
-			t.Errorf("unexpected error: %v", err)
-		}
+		var response *http.Response
+		func() {
```
Comment (hzxuzhonghu) [inline staging/src/k8s.io/apiserver/pkg/server/filters/cors_test.go]: only modify here

File: staging/src/k8s.io/apiserver/pkg/server/filters/cors_test.go [hunk 2]
```diff
@@ -131,28 +133,31 @@ func TestCORSAllowedMethods(t *testing.T) {
...
-		defer server.Close()
-		client := http.Client{}
-
-		request, err := http.NewRequest(test.method, server.URL+"/version", nil)
-		if err != nil {
-			t.Fatalf("unexpected error: %v", err)
-		}
-		request.Header.Set("Origin", "allowed.com")
+		var response *http.Response
+		func() {
```
Comment (hzxuzhonghu) [inline staging/src/k8s.io/apiserver/pkg/server/filters/cors_test.go]: and here

**Comment by hzxuzhonghu** [comment]
@sttts @dims PTAL

**Comment by hzxuzhonghu** [comment]
@sttts PTAL

File: staging/src/k8s.io/apiserver/pkg/server/filters/cors_test.go [hunk 3]
```diff
@@ -131,28 +133,31 @@ func TestCORSAllowedMethods(t *testing.T) {
...
+			}
 
-		methodAllowed := allowsMethod(response, test.method)
-		switch {
-		case test.allowed && !methodAllowed:
-			t.Errorf("Expected %v to be allowed, Got only %#v", test.method, response.Header.Get("Access-Control-Allow-Methods"))
-		case !test.allowed && methodAllowed:
-			t.Errorf("Unexpected allowed method %v, Expected only %#v", test.method, response.Header.Get("Access-Control-Allow-Methods"))
-		}
+			methodAllowed := allowsMethod(response, test.method)
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/filters/cors_test.go]: Move this block out of the func? We don't use the response var outside of the func now.

Comment (hzxuzhonghu) [reply staging/src/k8s.io/apiserver/pkg/server/filters/cors_test.go]: make sense

**Comment by hzxuzhonghu** [comment]
@sttts comment addressed.

**Pull request merged**